### ISSUE TITLE
Normalizing issues

### ISSUE THREAD

**Issue by hotrush** [balazscsaba2006/socially#1]
Hi @balazscsaba2006, i have similar need in validation and normalization social accounts. Tried your library and received lot of wrong results. Try to add next rows into your ParserTest:getNormalizeUrls, nothing passed (empty string means errors or wrong account links):

```
['http://www.twitter.com/codeschool/dasd?dsadasd', 'https://twitter.com/codeschool'],
['https://www.twitter.com/codeschool?dasdad', 'https://twitter.com/codeschool'],
['https://www.twitter.com/Codeschool?dasdad', 'https://twitter.com/codeschool'],
['http://twitter.com/codeschool/dasd?dsadasd', 'https://twitter.com/codeschool'],
['https://twitter.com/codeschool?dasdad', 'https://twitter.com/codeschool'],
['https://twitter.com/#!/codeschool?dasdad', 'https://twitter.com/codeschool'],
['https://twitter.com/@codeschool?dasdad', 'https://twitter.com/codeschool'],
['https://twitter.com/code-school', 'https://twitter.com/code'],
['https://twitter.com/code_school', 'https://twitter.com/code_school'],
['https://twitter.com/greentextbooks#', 'https://twitter.com/greentextbooks'],
['https://twitter.com/shopbop#cs=ov=73421773243,os=1,link=footerconnecttwitterlink\',', 'https://twitter.com/shopbop'],
['https://twitter.com/share', ''],

['https://www.facebook.com/dizzain/?pnref=lhc', 'https://www.facebook.com/dizzain'],
['http://www.facebook.com/dizzain?pnref=lhc', 'https://www.facebook.com/dizzain'],
['http://facebook.com/dizzain?pnref=lhc', 'https://www.facebook.com/dizzain'],
['https://facebook.com/dizzain?pnref=lhc', 'https://www.facebook.com/dizzain'],
['https://facebook.com/Dizzain?pnref=lhc', 'https://www.facebook.com/dizzain'],
['https://www.facebook.com/pages/fasdfadsfasdfsadf/126287147568059?pnref=lhc', 'https://www.facebook.com/pages/fasdfadsfasdfsadf/126287147568059'],
['https://www.facebook.com/PHPtoday-1025912177431644/?fref=ts', 'https://www.facebook.com/phptoday-1025912177431644'],
['http://www.facebook.com/pages/The-bloomtrigger-project/125218650866978/fasdfas?asdas', 'https://www.facebook.com/pages/the-bloomtrigger-project/125218650866978'],
['http://www.facebook.com/pages/DealMarket/157833714232772', 'https://www.facebook.com/pages/dealmarket/157833714232772'],
['http://www.facebook.com/', ''],
['http://www.facebook.com/#!/pages/dealmarket/157833714232772', 'https://www.facebook.com/pages/dealmarket/157833714232772'],
['http://www.facebook.com/home.php?tests#!/pages/dealmarket/157833714232772', 'https://www.facebook.com/pages/dealmarket/157833714232772'],
['http://www.facebook.com/pages/san-diego-ca/layer3-security-services/207635209271099', 'https://www.facebook.com/pages/san-diego-ca/layer3-security-services/207635209271099'],
['http://www.facebook.com/pages/agility+inc./114838698562760', 'https://www.facebook.com/pages/agility+inc./114838698562760'],
['http://www.facebook.com/pages/karen-mali-m%c3%bc%c5%9favirlik-logo-muhasebe/194296120603783', 'https://www.facebook.com/pages/karen-mali-m%c3%bc%c5%9favirlik-logo-muhasebe/194296120603783'],
['http://www.facebook.com//pages/custom-case-guy/1445342082363874', 'https://www.facebook.com/pages/custom-case-guy/1445342082363874'],
['https://www.facebook.com/sharer/sharer.php?u=asdasd', ''],
['https://www.facebook.com/share.php?u=asdasd', ''],
```

What do you think?

**Comment by balazscsaba2006** [comment]
Hi @hotrush. Made a commit on branch https://github.com/balazscsaba2006/socially/tree/issues/1 with updated tests, however it was mostly working as should.

Just to be clear, the normalization process removes unneded parts of a URL, meaning it strips subdomains (except for StackOverflow and Google Plus) including the `www.` part, and also removes query strings and fragments from the URL. Take a look at `DefaultNormalizer.php` to better understand how it works.

**Comment by balazscsaba2006** [comment]
What the commit on `issues/1` branch does is to clean `sharer` URLs from Twitter and Facebook, and strip out hashbangs and at-mentions from Twitter.

**Comment by balazscsaba2006** [comment]
If your use-case is different or you want the support for more platforms or different URLs, just make a pull request and we'll start from there, for eg. to add support for Facebook pages.

**Comment by balazscsaba2006** [comment]
Closed by #2 .
